src/backend/strategies.rs

```diff
@@ -1148,9 +1148,9 @@ pub async fn run_strategy_task<'s>(
                 let mut rng = StdRng::from_entropy(); // Will be passed to state_transitions_for_block
                 let mut current_block_info = initial_block_info.clone(); // Used for transition creation and logging
                 let mut transition_count: u32 = 0; // Used for logging how many transitions we attempted
-                let mut success_count: u32 = 0; // Used for logging how many transitions were successful
+                let mut success_count = Arc::new(AtomicU32::new(0)); // Used for logging how many transitions were successful
```

Comment:
okay since you wrote it and not Ivan, maybe this time I'll learn why `Arc` is used?